Methods comparing corpora (#3)
greenelab/annorxiver_manuscript: content/03.methods.md

2. Describe metadata statistics on PubMed central (number of articles, how many articles were processed

### Comparing Corpora

> We used gensim [@raw:rehurek_lrec] (version 3.8.1) to perform the above preprocessing tasks on the bioRxiv and PubMed Central corpora.

Reviewer: Did something get lost here? I'm not sure what was above.

Author: I initially wrote in the methods section that I used the spaCy package for pre-processing. This wasn't the case as I mixed up this project with another project that consisted of inserting Pubtator Central into a postgres database (in preparation of turning around the text_mining_hetnet_manuscript). 

The changed part here is that the annorxiver project only uses the gensim package to remove stop words and I _didn't_ perform any lemmatization (converting words into their root form).

> We used gensim [@raw:rehurek_lrec] (version 3.8.1) to perform the above preprocessing tasks on the bioRxiv and PubMed Central corpora.

Reviewer: ```suggestion
We used gensim [@raw:rehurek_lrec] (version 3.8.1) to preprocess the bioRxiv and PubMed Central corpora.
```
Is this correct? "the above preprocessing tasks" remains really confusing to me, since I don't think there's anything above this.

Author: Yeah that's correct. I was suppose to remove "the above" since the intended goal was for this sentence to come first.

> Gensim provides a listing of 337 unique stopwords ranging from 'I' to 'wherever' and we removed these words by using gensim's remove_stopwords function.

Reviewer: ```suggestion
We removed the 337 gensim-provided stopwords.
```

2. Describe metadata statistics on PubMed central (number of articles, how many articles were processed

### Comparing Corpora

> The first step in linguistic analysis is to preprocess the text for ease of computation.

Reviewer: ```suggestion

```

> This step consists of removing non-informative words (i.e., stop words) as well as converting words into their root form (i.e., lemmatization).

Reviewer: ```suggestion
We removed stop words _HOW,WHICH ONES_ and converted the remaining words into their root form (i.e., lemmatization) _HOW_.
```

> Throughout our analysis we encountered symbols that aren't considered words (e.g., $\pm$), so we will refer to words and symbols as tokens to avoid confusion.

Reviewer: ```suggestion
Throughout our analysis we encountered non-word symbols (e.g., $\pm$), so we refer to words and symbols as tokens to avoid confusion.
```

> We used spaCy [@raw:honnibal2017spacy] to perform the above preprocessing tasks on the bioRxiv and PubMed Central corpora.

Reviewer: I see now - recommend putting this front and center in the first sentence as th emethod 😁

Following the cleaning process, we calculated the frequency of each token between both corpora.

> Due to the large influx of unique tokens, we subsampled our analysis to use the union of the top 100 most frequent tokens between both corpora.

Reviewer: ```suggestion
Because many tokens were unique to one set or the other and observed at low frequency, we used the union of the top 100 most frequent tokens from each corpus to compare them.
```

> For each token we generated a contingency table as described in [@doi:10.1075/ijcl.6.1.05kil] (Table {@tbl:term-freq-contingency-table}).

Reviewer: I'm not sure that you need a description of how to make a contingency table in this research paper, though this text could be good for a review (maybe in the future but not now 😁 ). What do you think about removing the table and simplifying this to 
```suggestion
For each token we generated a contingency table and calculated the odds ratio for each of these tokens.
```

Then this would be joined with the portion below.

| | corpus X | corpus Y | |
|---|---|---|---|
| w | a | b | a+b |
| $\neg$w | c | d | c+d |
| | a+c | b+d | a+b+c+d = Total |

Table: A contingency table for a given token that may appear in two corpora that are being compared.
Given a query token $w$, the first row depicts the frequency of $w$ appearing in corpus X (a) and corpus Y (b).
The second row depicts the frequency of all other tokens appearing in corpus X (c) and corpus Y (d).
The total number of tokens can be found by adding (a,b,c,d) together as depicted in the bottom right.
{#tbl:term-freq-contingency-table}

Once the contingency table has been created, we calculated the odds ratio  for a given token.

> The odds ratio is defined as the fraction of the probability of a token $w$ appearing in corpus X divided by the probability of a token $w$ appearing in corpus Y.

Reviewer: This is also more introductory than I think you want in this paper. Similar to above I would recommend removing this definition, but one day for an introductory review this would be great to have :)

Some tokens may only appear in one corpus and not another, which may result in a non-real number.

> Our fix to this problem is to add a small epsilon ($1\mathrm{e}{-20}$) to every value in the contingency table.

Reviewer: Did you have to do this within the top 100 tokens? I would be surprised.

Derivation for the odds ratio is provided here:

> $$OR= \frac{\frac{a}{c}}{\frac{b}{d}}=\frac{a*d}{b*c}$$

Reviewer: Similar - more than needed.

The range of this ratio is $\[0, \infty\)$ with numbers greater than one implying that there is a strong association with a token appearing in corpus X compared to corpus Y.
Contrariwise, numbers less than one imply that there is a strong association with a token appearing in corpus Y rather than corpus X.

> We calculated 95% confidence intervals for each odds ratio as mentioned in [@https://www.ncbi.nlm.nih.gov/books/NBK431098/].

Reviewer: I do think it's good to describe the procedure you use to calculate the 95% CIs w/ a reference but the entire formula / procedure is likely too much for this paper.
